spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/EnableAutoConfigurationImportSelector.java

```diff
@@ -205,6 +206,22 @@ private MetadataReaderFactory getMetadataReaderFactory() {
 		}
 	}
 
+	private void checkExcludedClasses(List<String> configurations,
+			Set<String> exclusions) {
+		StringBuilder message = new StringBuilder();
+		for (String exclusion : exclusions) {
+			if (!configurations.contains(exclusion)) {
+				message.append("\t- ").append(exclusion).append("\n");
+			}
+		}
+		if (!message.toString().isEmpty()) {
+			throw new IllegalStateException(
+					"Could not exclude the following classes from autoconfiguration:\n"
```

Comment:
yeah I think having a dedicated one might be an overkill but adding an explicit message is a good idea. I had it in there but it seemed too long. I'll put it back.